spec/models/manageiq/providers/lenovo/physical_infra_manager/refresher_spec.rb

```diff
@@ -17,142 +32,631 @@
+        # Perform the refresh twice to verify that a second run with existing data
+        # does not change anything
+        2.times do
+          EmsRefresh.refresh(ems2)
+          ems2.reload
 
-  def assert_specific_storage
-    storage = PhysicalStorage.find_by(:ems_ref => "208000C0FF2683AF")
-
-    expect(storage.name).to eq("S3200-1")
-    expect(storage.uid_ems).to eq("208000C0FF2683AF")
-    expect(storage.ems_ref).to eq("208000C0FF2683AF")
-    expect(storage.access_state).to eq("Online")
-    expect(storage.access_state).to eq("Online")
-    expect(storage.health_state).to eq("Critical")
-    expect(storage.overall_health_state).to eq("Critical")
-    expect(storage.type).to eq("ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalStorage")
-    expect(storage.drive_bays).to eq(12)
-    expect(storage.enclosures).to eq(1)
-    expect(storage.canister_slots).to eq(2)
-  end
+          assert_table_counts(:ems2)
+          assert_specific_rack(:ems2)
+          assert_specific_server(:ems2)
+          assert_specific_storage(:ems2)
+          assert_specific_switch(:ems2)
+          assert_guest_table_contents
+          assert_physical_network_ports_table_content
+          assert_physical_network_ports_connection
+        end
+      end
 
-  def assert_table_counts
-    expect(PhysicalRack.count).to eq(3)
-    expect(PhysicalServer.count).to eq(2)
-    expect(PhysicalStorage.count).to eq(2)
-    expect(GuestDevice.count).to eq(8)
-    expect(PhysicalNetworkPort.count).to eq(50)
-  end
+      def lenovo_model(model)
+        name = model.class.to_s.split('::').last
 
-  def assert_guest_table_contents
-    server = PhysicalServer.find_by(:ems_ref => "7936DD182C5311E3A8D6000AF7256738")
-    nic = server.hardware.nics.first
-    expect(nic.device_name).to eq("Broadcom 2-port 1GbE NIC Card for IBM")
-  end
+        lenovo_klass = "ManageIQ::Providers::Lenovo::PhysicalInfraManager::#{name}".safe_constantize
+        if lenovo_klass.present? && lenovo_klass.to_s.split('::').size > 1
+          model.becomes(lenovo_klass)
+        else
+          model
+        end
+      end
 
-  def assert_physical_network_ports_table_content
-    server = PhysicalServer.find_by(:ems_ref => "7936DD182C5311E3A8D6000AF7256738")
-    ports = server.hardware.nics.first.physical_network_ports
+      def assert_table_counts(manager = :ems1)
+        if manager == :ems1
+          expect(PhysicalRack.count).to eq(1)
+          expect(PhysicalServer.count).to eq(3)
+          expect(PhysicalStorage.count).to eq(3)
+          expect(PhysicalChassis.count).to eq(1)
+          expect(PhysicalSwitch.count).to eq(1)
+          expect(AssetDetail.count).to eq(8)
+          expect(PhysicalDisk.count).to eq(12)
+          expect(Canister.count).to eq(6)
+          expect(ComputerSystem.count).to eq(10)
+          expect(Hardware.count).to eq(11)
+          expect(GuestDevice.count).to eq(14)
+          # expect(Firmware.count).to eq(20)
```

Comment:
@douglasgabriel This line is not true for first round of old refresh. It's equal only in 2nd round and both rounds of graph refresh

Reply:
```suggestion
         expect(Firmware.count).to eq(20)
```
Please rebase this PR and remove the `#`

Reply:
@CharlleDaniel done